### PR TITLE
[AWS Lambda] Adding timeout for each resource

### PR DESCRIPTION
**Pull request by binodmx**
Screenshot from 2019-12-03 21-44-52

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/AWSLambdaMediator.java [hunk 1]
```diff
@@ -121,13 +123,19 @@ private InvokeResult invokeLambda(String payload) {
                 log.error("Missing AWS Credentials");
                 return null;
             }
-            AWSLambda awsLambda = AWSLambdaClientBuilder.standard()
-                    .withCredentials(credentialsProvider)
-                    .build();
+            // set invoke request
+            if (resourceTimeout < 1000 && resourceTimeout > 900000) {
```
Comment (Krishanx92) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/AWSLambdaMediator.java]: Could this be null?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 2]
```diff
@@ -574,10 +574,13 @@ private void updateOperationManagedInfo(SwaggerData.Resource resource, Operation
         }
         operation.setVendorExtension(APIConstants.SWAGGER_X_AUTH_TYPE, authType);
         operation.setVendorExtension(APIConstants.SWAGGER_X_THROTTLING_TIER, resource.getPolicy());
-        // AWS Lambda: set arn to swagger
+        // AWS Lambda: set arn & timeout to swagger
         if (resource.getAmznResourceName() != null) {
             operation.setVendorExtension(APIConstants.SWAGGER_X_AMZN_RESOURCE_NAME, resource.getAmznResourceName());
         }
+        if (resource.getAmznResourceTimeout() != 0) {
```
Comment (Krishanx92) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java]: What happen if resource.getAmznResourceTimeout() null?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/AWSLambdaMediator.java (hunk 1 above)
Comment (binodmx) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/AWSLambdaMediator.java]: It's an integer. If not defined it'll be 0.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java (hunk 2 above)
Comment (binodmx) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java]: if not initialized it'll be 0

**Pull request merged**